ast/src/parsed/mod.rs

```diff
@@ -395,6 +395,15 @@ pub struct TraitFunction<E = u64> {
     pub ty: Type<E>,
 }
 
+impl TraitFunction<u64> {
+    pub fn type_scheme(&self, vars: TypeBounds) -> TypeScheme<u64> {
```

Comment:
Wasn't there a struct where you also have a reference to the trait (so you could get the vars from there)?
If not I'm sorry, this way the function is not too useful...